src/v/kafka/server/connection_context.cc

```diff
@@ -231,6 +228,10 @@ ss::future<> connection_context::stop() {
     co_await _gate.close();
     co_await _as.stop();
 
+    if (_hook && is_linked()) {
```

Comment:
There is a another use of this list, presumably, and that's on shutdown. I think `stop()` can be a race as the unlinking is now asynchronous (it's after a co_await), so this needs protecting against double `stop()`.